### ISSUE TITLE
ODEINT Controlled Steppers do not work with complex thrust device vectors as state type.

### ISSUE THREAD

**Issue by Vishwatosh** [headmyshoulder/odeint-v2#252]
I found that the controlled steppers of ODEINT are not working when the **_state_type_** is a thrust::device_vector<thrust::complex<double>>. Does anyone else have the same problem. It would be very helpful if anyone can please suggest a possible solution for the problem?

**Comment by anates** [comment]
I solved it by recasting the vector into thrust::device_vector<double> before transferring it to the odeint-function. In my RHS-function I casted it back into thrust::device_vector<thrust::complex>. I know that that is not the best approach, but at least a workaround.

**Comment by Vishwatosh** [comment]
Hi Anates,
Thanks for your reply. As far as I understand from your reply that you convert the device_vector of <thrust::complex> to the device_vector of <double> (something like interleaved real and imaginary parts in the new device vector) before entering the system function of the ODEINT and then convert the final result back to a device_vector of <thrust::complex>? Could you kindly confirm if my understanding is correct. I am asking this because I have used this method as a workaround.

Thanks a lot!

**Comment by anates** [comment]
Yes, more or less like that. Before entering the boost::odeint-function I recasted it into a <double>-vector, and within my right hand side function it was casted back into a thrust::complex-vector. The return-value of the RHS was re-casted into double, and casted back into thrust::complex afterwards
